flag_generic.go

```diff
@@ -0,0 +1,67 @@
+package cli
+
+type GenericFlag = FlagBase[Value, NoConfig, genericValue]
```

Comment:
I think `genericValue` should be made public as mixed type access in a public API can be no fun.

Reply:
Not really since the public API would use only ```cli.Value``` . Any other customizations is upto the user to redefine FlagBase with other template types